### PR TITLE
[cr133] Remove all `NOTREACHED_IN_MIGRATION` remaining

### PR DESCRIPTION
**Pull request by cdesouza-chromium**
`NOTREACHED_IN_MIGRATION` has now been deleted from chromium in M133. This was expected, and the existing occurrances have to be migrated to `NOTRECHEAD`. At this point, the only difference between `NOTREACHED`, and `NOTREACHED_IN_MIGRATION`, is that `NOTREACHED` is tagged as no return.

Chromium change:
https://chromium.googlesource.com/chromium/src/+/e238462dc67196d361ee0bbaf1954969619cab27

```
commit e238462dc67196d361ee0bbaf1954969619cab27
Author: Peter Boström <pbos@chromium.org>
Date:   Fri Nov 15 00:46:51 2024 +0000

    Remove NOTREACHED_IN_MIGRATION()

    Bug: 40580068
```

Resolves https://github.com/brave/brave-browser/issues/42330

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: browser/brave_shields/brave_shields_tab_helper.cc [hunk 1]
```diff
@@ -260,8 +260,7 @@ CookieBlockMode BraveShieldsTabHelper::GetCookieBlockMode() {
     case ControlType::DEFAULT:
       break;
   }
-  NOTREACHED_IN_MIGRATION();
-  return CookieBlockMode::BLOCKED;
+  NOTREACHED();
```
Comment (tmancey) [inline browser/brave_shields/brave_shields_tab_helper.cc:263]: nit: `NOTREACHED() << "Unexpected value for control_type: "<< base::to_underlying(control_type);`

File: browser/metrics/metrics_reporting_util.cc [hunk 2]
```diff
@@ -26,10 +26,8 @@ bool GetDefaultPrefValueForMetricsReporting() {
       return true;
     case version_info::Channel::UNKNOWN:
       return false;
-    default:
-      NOTREACHED_IN_MIGRATION();
-      return false;
   }
+  NOTREACHED();
```
Comment (tmancey) [inline browser/metrics/metrics_reporting_util.cc:30]: nit: `NOTREACHED() << "Unexpected value for channel: "<< base::to_underlying(channel);`

File: chromium_src/third_party/blink/renderer/modules/plugins/dom_plugin_array.cc [hunk 3]
```diff
@@ -136,11 +136,10 @@ void FarblePlugins(DOMPluginArray* owner,
       dom_plugins->push_back(fake_dom_plugin_2);
       // Shuffle the list of plugins pseudo-randomly, based on the domain key.
       std::shuffle(dom_plugins->begin(), dom_plugins->end(), prng);
-      break;
+      return;
     }
-    default:
-      NOTREACHED_IN_MIGRATION();
   }
+  NOTREACHED();
```
Comment (tmancey) [inline chromium_src/third_party/blink/renderer/modules/plugins/dom_plugin_array.cc:142]: nit: `NOTREACHED() << "Unexpected value for farbling_level: "<< base::to_underlying(farbling_level);`

**Review by tmancey — APPROVED** [review]
LGTM++ (just a couple nits)

File: browser/brave_shields/brave_shields_tab_helper.cc (hunk 1 above)
Comment (cdesouza-chromium) [reply browser/brave_shields/brave_shields_tab_helper.cc]: Done

File: browser/metrics/metrics_reporting_util.cc (hunk 2 above)
Comment (cdesouza-chromium) [reply browser/metrics/metrics_reporting_util.cc]: Done

**Review by yrliou — APPROVED** [review]
wallet++

**Review by goodov — APPROVED** [review]

**Comment by bsclifton** [comment]
Good to go - admin merging

**Pull request merged**